solutions/CSES Planets Cycles.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: cses-1751
+source: CSES 
+title: Planets Cycles
+author: Maggie Liu
+---
+
+Define the number of teleportations starting from a planet as the $\texttt{cyclelength}$ of that planet. For each planet that hasn't been visited, we want to find its $\texttt{cyclelength}$. Call the planet we are performing $\texttt{dfs}$ from the $\textit{start}$. As we perform $\texttt{dfs}$ from the $\textit{start}$, keep track of the planets seen, in order, in the $\texttt{cycle}$ queue. When we reach a planet that has already been visited (call this planet the $\textit{repeat}$), we add the $\texttt{cyclelength}$ of the $\textit{repeat}$ to the current $\texttt{step}$ count. 
+
+Once we have $\texttt{cycle}$ and $\texttt{steps}$ (the $\texttt{cyclelength}$ of the $\textit{start}$) we can calculate the $\texttt{cyclelength}$ of all the planets in this $\texttt{cycle}$. We know the $\textit{repeat}$ will always be the planet at the end of the $\texttt{cycle}$, but there are two cases for the $\textit{repeat}$. 
+1. The $\textit{repeat}$ was visited twice in the current $\texttt{cycle}$.
+2. The $\textit{repeat}$ was visited previously and only appears at the end of the current $\texttt{cycle}$.
+
+For the first case, let the planets in the $\texttt{cycle}$ between the two occurrences of the $\textit{repeat}$ be the $\textit{loop}$. For the planets inside the $\textit{loop}$, the repeating planet when starting from that planet is itself. Thus, the $\texttt{cyclelength}$ of any planet inside the $\textit{loop}$ will be the same as the $\texttt{cyclelength}$ of the $\textit{repeat}$. For all the planets outside the $\textit{loop}$, the planet that repeats when starting from each planet will still be the $\textit{repeat}$.
```

Comment:
cycle having equal len can be explained separately below. Here, just mention that the nodes between two instances of repeat form a cycle. In fact, this is so short it can probably be put into the ordered list.